mitmproxy/addons/modifyheaders.py

```diff
@@ -2,28 +2,29 @@
 
 from mitmproxy import exceptions
 from mitmproxy import flowfilter
+from mitmproxy.utils import strutils
 from mitmproxy import ctx
 
 
-def parse_modify_headers(s):
+def parse_modify_hook(s):
     """
-        Returns a (header_name, header_value, flow_filter) tuple.
+        Returns a (flow_filter, header_name, header_value) tuple.
```

Comment:
If we re-use this (which is good 👍), we should probably signal somehow that it's not just headers. We can also add a bit more type information to make things easier to grasp:

```python
class ModifySpec(typing.NamedTuple):
    matches: flowfilter.TFilter,
    match_str: str
    key: bytes  # those could potentially have better names if you can think of some
    replacement: bytes


def parse_modify_spec(s: str) -> ModifySpec:
    # ...
    filter = flowfilter.parse(filter_str)
    return ModifySpec(filter, filter_str, header_name, header_value)
```

then, below:
```python
class ModifyHeaders:
    def __init__(self):
        replacements: List[ModifySpec] = []    # replaces self.lst
    
    # ...
    
    def run(self, f, hdrs):
        for spec in self.replacements:
            if spec.matches(f):
                hdrs.pop(spec.key, None)
```

I realize this is just a very rough sketch. Does it make sense anyway?

Reply:
Great idea, thanks 🙌